### PR TITLE
feat: add admin roles which have admin control over a graphql api

### PR DESCRIPTION
**Pull request by SwaySway**
#### Description of changes
- extend admin ui role check to work with functions which have access to graphql apis
- updated unit/e2e
- change the options for the auth transformer class to only require an admin role list to determine if the iam directives should added on a `@model`

#### Description of how you validated changes
- unit/e2e

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by lazpavel — APPROVED** [review]
seems to have build errors

**Review by lazpavel — APPROVED** [review]
LGTM, can you fix the bot alerts

**Comment by SwaySway** [comment]
bot alerts addressed

File: packages/amplify-graphql-auth-transformer/src/resolvers/query.ts [hunk 1]
```diff
@@ -303,7 +312,8 @@ export const generateAuthExpressionForRelationQuery = (
   fields: ReadonlyArray<FieldDefinitionNode>,
   primaryFieldMap: RelationalPrimaryMapConfig,
 ) => {
-  const { cogntoStaticRoles, cognitoDynamicRoles, oidcStaticRoles, oidcDynamicRoles, apiKeyRoles, iamRoles, lambdaRoles } = splitRoles(roles);
+  const { cogntoStaticRoles, cognitoDynamicRoles, oidcStaticRoles, oidcDynamicRoles, apiKeyRoles, iamRoles, lambdaRoles } =
```
Comment (cjihrig) [inline packages/amplify-graphql-auth-transformer/src/resolvers/query.ts:315]: Doesn't look related to this PR, but is `cogntoStaticRoles` a typo?

**Review by cjihrig — APPROVED** [review]

File: packages/amplify-graphql-auth-transformer/src/resolvers/query.ts (hunk 1 above)
Comment (SwaySway) [reply packages/amplify-graphql-auth-transformer/src/resolvers/query.ts:315]: Yup 😅 I can address that here as well
